research/README.md

```diff
@@ -0,0 +1,325 @@
+
+**Desktop** lists files by filename, sorted with most recently added at the top. The time the file was added is provided as muted, secondary text. In the 0.3, you can't drill down into folders, though I think this feature is available in the latest dev version.
+
+**Companion** has no file browsing feature.
+
+## Connections / Peers
+
+Desktop and WebUI both have a page for showing info about all the other IPFS nodes you are currently connected to.
+
+| webui | desktop |
+|-------|---------|
+| ![](img/003-peers-webui-1.png) | ![](img/003-peers-desktop-1.png) |
+
+**WebUI** shows a peer count, geolocated pins on a globe and list of Peer Id / Multiaddr pairs for each connection. Clicking on a list item expands it to show similar node info as show on the Home page about your local node. The process is resource intensive and causes "slow script" warnings in firefox.
+
+**Desktop** Shows your location and peer count, along with a list of Peer Id / Location pairs. No more info is available about the peers. There is a search bar that allows you to search for peers by id or location.
+
+**Companion** shows just the number of connected peers as a menu bar stat, and again when you go to share a file.
+
+## DAG / IPLD Explorer
+
+Discussion: https://github.com/ipfs-shipyard/pm-ipfs-gui/issues/11
+
+**WebUI** has a DAG explorer, which might be due a rename to "IPLD Explorer". It takes an IPFS hash (also due a rename to CID) and lists out all the links or data that in the object the CID points to. In the case of a directory, you can see a list of the files it contains, and each filename is a link to it's IPLD object, allowing you drill down, navigating through the links. Where a hash points to data, the raw data is rendered in plain text.
+
+| Empty state | Explore a cid | Drill into a data node |
+|-|-|-|
+|![](img/004-ipld-explorer-webui-1.png) |![](img/004-ipld-explorer-webui-2.png) |![](img/004-ipld-explorer-webui-3.png)
+
+The UI/UX should be improved but the idea is interesting, and could become a vital tool for helping people to understand IPFS and IPLD.
+
+## Pinning
+
+Discussion: https://github.com/ipfs-shipyard/pm-ipfs-gui/issues/10
+
+**Desktop** allows you to pin a CID. You don't need to aleady store that content. You can paste in any CID and give it a label to remind yourself what it was.
+
+| Desktop pin pane | Add a pin |
+|------------------|-----------|
+| ![](img/005-pin-desktop-1.png) | ![](img/005-pin-desktop-2.png) |
+
+| Filled out form | PIN in progesss
+|-----------------|------------------|
+| ![](img/005-pin-desktop-3.png) | ![](img/005-pin-desktop-4.png) |
+
+## Open WebUI
+
+Both Desktop and Companion offer this, but is "Open WebUI" the right name for this action? As a companion or desktop user, I may have never seen the WebUI app, and so may have no intuitions as to what it's for.
+
+Desktop and companion have differing ideas as to what the best url for WebUI is.
```

Comment:
I think companion uses IP for API just for visual consistency with Gateway.
We have to use IP instead of `localhost` for Gateway due to: https://github.com/ipfs-shipyard/ipfs-companion/issues/328